core/src/price_estimation/clients/oneinch/api.rs

```diff
@@ -0,0 +1,179 @@
+use super::super::generic_client::{Api, GenericToken};
+use crate::http::{HttpClient, HttpFactory, HttpLabel};
+use anyhow::{Context, Result};
+use ethcontract::Address;
+use futures::future::{BoxFuture, FutureExt as _};
+use serde::Deserialize;
+use std::collections::HashMap;
+use url::Url;
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct Token {
+    pub name: String,
+    pub symbol: String,
+    pub address: Option<Address>,
+    pub decimals: u8,
+}
+impl GenericToken for Token {
+    fn symbol(&self) -> &str {
+        &self.symbol
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct TradedAmounts {
+    to_token_amount: String,
```

Comment:
nit: I think that the traded amounts should be parsed at deserialization time, since `{"toTokenAmoun":"blabla"}` should fail to deserialize.
```suggestion
    #[serde(with = "display_fromstr")]
    to_token_amount: u128,
```

This also requires:
```rust
use serde_with::rust::display_fromstr;
```

Reply:
Done! I parsed it as `f64` instead of `u128` since there is no `.into()` or any easy conversion I could find from `u128` to `f64`.

Reply:
You should be able to do a numeric cast with `as`: `100u128 as f64`.

Reply:
I think it would be *slightly* more correct to parse into `u128` since the values are atoms, which would avoid rounding errors for large numbers (although to be fair, there would be rounding errors when computing the exchange rate, so ultimately it makes no difference).